pallets/pallet-asset-switch/src/xcm/trade/switch_pair_remote_asset/tests/drop.rs

```diff
@@ -0,0 +1,136 @@
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+use frame_support::{assert_storage_noop, traits::fungible::Inspect as InspectFungible};
+use sp_core::Get;
+use sp_runtime::{
+	traits::{One, Zero},
+	AccountId32,
+};
+use xcm::v3::{Junction, Junctions, MultiLocation};
+use xcm_executor::traits::WeightTrader;
+
+use crate::xcm::{
+	test_utils::get_switch_pair_info_for_remote_location_with_pool_usable_balance,
+	trade::{
+		switch_pair_remote_asset::mock::{Balances, ExtBuilder, MockRuntime, ToDestinationAccount},
+		test_utils::SumTimeAndProofValues,
+	},
+	UsingComponentsForSwitchPairRemoteAsset,
+};
+
+#[test]
+fn happy_path() {
+	let location = MultiLocation {
+		parents: 1,
+		interior: Junctions::X1(Junction::Parachain(1_000)),
+	};
+	// ED + 1
+	let new_switch_pair_info =
+		get_switch_pair_info_for_remote_location_with_pool_usable_balance::<MockRuntime>(&location, 1);
+	ExtBuilder::default()
+		.with_switch_pair_info(new_switch_pair_info.clone())
+		.build_and_execute_with_sanity_tests(|| {
+			let weigher = {
+				let mut weigher = UsingComponentsForSwitchPairRemoteAsset::<
+					MockRuntime,
+					_,
+					SumTimeAndProofValues,
+					ToDestinationAccount,
+				>::new();
+				weigher.remaining_fungible_balance = 1;
+				weigher
+			};
+			assert!(<Balances as InspectFungible<AccountId32>>::balance(&ToDestinationAccount::get()).is_zero());
+			drop(weigher);
+			assert!(<Balances as InspectFungible<AccountId32>>::balance(&ToDestinationAccount::get()).is_one());
+			assert!(<Balances as InspectFungible<AccountId32>>::balance(&new_switch_pair_info.pool_account).is_one());
+		});
+}
+
+#[test]
+fn no_switch_pair() {
+	ExtBuilder::default().build_and_execute_with_sanity_tests(|| {
+		let weigher = {
+			let mut weigher = UsingComponentsForSwitchPairRemoteAsset::<
+				MockRuntime,
+				_,
+				SumTimeAndProofValues,
+				ToDestinationAccount,
+			>::new();
+			weigher.remaining_fungible_balance = 1;
+			weigher
+		};
+		assert_storage_noop!(drop(weigher));
+	});
+}
+
+#[test]
+fn zero_remaining_balance() {
+	let location = MultiLocation {
+		parents: 1,
+		interior: Junctions::X1(Junction::Parachain(1_000)),
+	};
+	let new_switch_pair_info =
+		get_switch_pair_info_for_remote_location_with_pool_usable_balance::<MockRuntime>(&location, 0);
+	ExtBuilder::default()
+		.with_switch_pair_info(new_switch_pair_info)
+		.build_and_execute_with_sanity_tests(|| {
+			let weigher = {
+				let mut weigher = UsingComponentsForSwitchPairRemoteAsset::<
+					MockRuntime,
+					_,
+					SumTimeAndProofValues,
+					ToDestinationAccount,
+				>::new();
+				weigher.remaining_fungible_balance = u128::zero();
+				weigher
+			};
+			assert_storage_noop!(drop(weigher));
+		});
+}
+
+#[test]
+#[should_panic(expected = "Transferring from pool account to fee destination failed.")]
```

Comment:
Because we are testing a condition which in production should never happen, which is that the pool account does not have enough balance. I added a `debug_assert` which is triggered in this specific test case, that's why the `should_panic`.